diesel/src/query_builder/upsert/on_conflict_clause.rs

```diff
@@ -3,41 +3,62 @@ use super::on_conflict_target::*;
 use crate::backend::sql_dialect;
 use crate::backend::Backend;
 use crate::insertable::*;
+use crate::query_builder::where_clause::NoWhereClause;
 use crate::query_builder::*;
 use crate::result::QueryResult;
 
 #[doc(hidden)]
 #[derive(Debug, Clone, Copy)]
-pub struct OnConflictValues<Values, Target, Action> {
+pub struct OnConflictValues<Values, Target, Action, WhereClause> {
     values: Values,
     target: Target,
     action: Action,
+    /// Allow to apply filters on ON CONFLICT ... DO UPDATE ... WHERE ...
+    where_clause: WhereClause,
 }
 
-impl<Values, Target, Action> QueryId for OnConflictValues<Values, Target, Action> {
+impl<Values, Target, Action, WhereClause> QueryId
+    for OnConflictValues<Values, Target, Action, WhereClause>
+{
     type QueryId = ();
 
     const HAS_STATIC_QUERY_ID: bool = false;
 }
 
-impl<Values> OnConflictValues<Values, NoConflictTarget, DoNothing> {
+impl<Values> OnConflictValues<Values, NoConflictTarget, DoNothing, NoWhereClause> {
     pub(crate) fn do_nothing(values: Values) -> Self {
-        Self::new(values, NoConflictTarget, DoNothing)
+        Self::new(values, NoConflictTarget, DoNothing, NoWhereClause {})
```

Comment:
```suggestion
        Self::new(values, NoConflictTarget, DoNothing, NoWhereClause)
```

as that's the idiomatic way to instance zero sized structs